### PR TITLE
fix: mark spans representing calls to external services as exit spans

### PR DESCRIPTION
**Pull request by kruskall**
Exit spans are spans that describe a call to an external service, such as an outgoing HTTP request or a call to a database.

Marking a span as an exit span also triggers service target fields and destination fields inference.
Update tests to reflect the new fields.

Disable compression in some tests as exit spans can be aggregated based on certain fields, leading to unexpected results.

Additionally, make sure exit spans are not being used as parents of other spans as that would cause the spans to be dropped.

Closes https://github.com/elastic/apm-agent-go/issues/1315

### CODE REVIEW

File: apmtest/withtransaction.go
Comment (axw) [inline apmtest/withtransaction.go:32]: nice :)

File: module/apmhttp/clienttrace.go
Comment (axw) [inline module/apmhttp/clienttrace.go:63]: The spec says:

> Exit spans MAY have child spans that have the same type and subtype. For example, an HTTP exit span may have child spans with the action request, response, connect, dns. These spans MUST NOT have any destination context, so that there's no effect on destination metrics.

If they mustn't have destination context, does that mean they should not be exit spans?

File: module/apmhttp/client.go
Comment (axw) [inline module/apmhttp/client.go:115]: See my other comment in clienttrace.go - I'm pretty sure this span should always be an exit span. Because the net/http/httptrace spans have the same type and subtype, only a different action, they can be children of the exit span.

File: module/apmgormv2/apmgorm_test.go
Comment (axw) [inline module/apmgormv2/apmgorm_test.go:126]:
```suggestion
		assert.Equal(t, dsnInfo.Database, span.Context.Destination.Service.Name)
```

Right?

**Review by axw — COMMENTED** [review]
Thanks, this is great. Just a few questions.

File: module/apmgormv2/apmgorm_test.go
Comment (kruskall) [reply module/apmgormv2/apmgorm_test.go]: This is actually destination service name, the old deprecated field. It should use the driver name.

File: module/apmhttp/client.go
Comment (kruskall) [reply module/apmhttp/client.go]: Good point! Updated!

File: module/apmhttp/clienttrace.go
Comment (kruskall) [reply module/apmhttp/clienttrace.go]: It is not specified but I think it makes sense for them not to be exit spans, especially considering the related features (compression, etc.)

**Comment by kruskall** [comment]
This uncovered another bug with the agent which is not respecting the spec.

Blocked by https://github.com/elastic/apm-agent-go/pull/1320

**Review by axw — COMMENTED** [review]
Thanks for the changes! LGTM, modulo the httptrace issue you uncovered.

**Comment by kruskall** [comment]
> This uncovered another bug with the agent which is not respecting the spec.
> 
> Blocked by #1320

PR merged. Unblocking

**Review by axw — APPROVED** [review]
LGTM. I think the integration tests need updating: https://apm-ci.elastic.co/blue/organizations/jenkins/apm-agent-go%2Fapm-agent-go-mbp/detail/PR-1317/8/pipeline

**Comment by axw** [comment]
There's a few cases where we're not marking spans as exit spans, and I found a panic related to ending a child of an exit span. I'll open a PR.

**Pull request merged**
